pkg/server/tenant_auto_upgrade.go

```diff
@@ -31,40 +31,41 @@ import (
 // starts. This is to cover cases where upgrade becomes possible due to
 // an upgrade to the tenant binary version.
 func (s *SQLServer) startTenantAutoUpgradeLoop(ctx context.Context) error {
-	storageClusterVersion := s.settingsWatcher.GetStorageClusterActiveVersion().Version
 	return s.stopper.RunAsyncTask(ctx, "tenant-auto-upgrade-checker", func(ctx context.Context) {
-		firstAttempt := true
+		loopFrequency := 30 * time.Second
+		if k := s.cfg.TestingKnobs.Server; k != nil {
+			loopFrequency = k.(*TestingKnobs).TenantAutoUpgradeLoopFrequency
+		}
+
 		for {
 			select {
 			case <-s.stopper.ShouldQuiesce():
 				return
-			// Check for changes every 10 seconds to avoid triggering an upgrade
-			// on every change to the internal version of storage cluster version
-			// within a short time period.
-			case <-time.After(time.Second * 10):
-				latestStorageClusterVersion := s.settingsWatcher.GetStorageClusterActiveVersion().Version
-				// Only run upgrade if this is the first attempt (i.e. on server startup) or if the
-				// the storage cluster version changed and is at an Internal version of 0 which implies that
-				// that storage is at the "final" version for some release. First case ensures that if an upgrade is
-				// possible due to a change in a sql instance binary version, it happens. Second
-				// cases ensures that if an upgrade is possible due to a change in the storage
-				// cluster version, it happens.
-				storageClusterVersionChanged := storageClusterVersion != latestStorageClusterVersion
-				if firstAttempt ||
-					(storageClusterVersionChanged && storageClusterVersion.Internal == 0) {
-					firstAttempt = false
-					storageClusterVersion = latestStorageClusterVersion
-					if err := s.startAttemptTenantUpgrade(ctx); err != nil {
+			case <-time.After(loopFrequency):
+				storageClusterVersion := s.settingsWatcher.GetStorageClusterActiveVersion().Version
+
+				// Only attempt to upgrade if the storage cluster is at an Internal version of 0,
+				// which implies that storage is at the "final" version for some release.
+				if storageClusterVersion.Internal == 0 {
+					log.Infof(ctx, "starting auto upgrade attempt")
```

Comment:
You've been working with this and have a better sense of what is useful, but I wonder if we want to remove this log message as we already have a number of them in startAttemptTenantUpgrade.

Reply:
Good point, this is not really needed or useful. Removed.